src/sentry/backup/imports.py

```diff
@@ -435,6 +440,47 @@ def resolve_org_slugs_from_pk_map(pk_map: PrimaryKeyMap):
             slug_mapping[org_id] = org_slug or ""
 
         if len(slug_mapping) > 0:
+            # HACK(azaslavsky): Okay, this gets a bit complicated, but bear with me: the following
+            # `bulk_create...` calls will result in some actions on the control silo that call back
+            # into this region. So the client (this region) calls the server (the control silo)
+            # which may need to make one or more calls back into the client (this region). Because
+            # some of our `OrganizationMemberTeam` outboxes may not be drained due to the HACK we
+            # performed in `import_export/impl.py` (see that file for more details), there may be a
+            # massive backlog of `OrganizationMemberTeam` outboxes that need to clear before this
+            # region can respond. In the worst case scenario, this will result in a timeout of the
+            # original, outer `bulk_create...` call.
+            #
+            # So, the solution we take here is to manually clear all `ORGANIZATION_SCOPE` outboxes
+            # for each imported organization on this side first, so that when this region responds
+            # to the call triggered from the server-side of `bulk_create...`, the
+            # `ORGANIZATION`-scoped outbox queue is empty and is free to only serve requests
+            # specifically related to slug provisioning.
+            for id in slug_mapping.keys():
```

Comment:
Is there potential for the slug mapping to contain more than 1 organization ID? If so, wouldn't the break below result in us skipping some organizations?

Reply:
If we import multiple orgs at a time (unlikely, though users are allowed to do this on ex: self-hosted), we'll have one entry per org imported. But the `break` should only apply to the inner loop, right?

Reply:
oh whoops, yes. My bad 😅